extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java

```diff
@@ -128,6 +133,77 @@ public class InfinispanClientRuntimeConfig {
     @ConfigItem
     Optional<String> trustStoreType;
 
+    /**
+     * Configures caches from the client with the provided configuration.
+     */
+    @ConfigItem
+    Map<String, RemoteCacheConfig> cache = new HashMap<>();
+
+    @ConfigGroup
+    public static class RemoteCacheConfig {
+
+        // @formatter:off
+        /**
+         * Cache configuration in inlined XML to create the cache on first access.
+         * Will be ignored if the configuration-file or the configuration-uri is provided for the same cache name.
+         * An example of the user defined property:
+         * quarkus.infinispan-client.cache.bookscache.configuration=<distributed-cache><encoding media-type="application/x-protostream"/></distributed-cache>
+         */
+        // @formatter:on
+        @ConfigItem
+        public Optional<String> configuration;
+
+        // @formatter:off
+        /**
+         * Cache configuration file in XML whose content will be used to create the cache on first access.
+         * Will be ignored if configuration-uri is provided.
+         *
+         * An example of the user defined property:
+         * quarkus.infinispan-client.cache.bookscache.configuration-file=cacheConfig.xml
+         */
+        // @formatter:on
+        @ConfigItem
+        public Optional<String> configurationFile;
+
+        // @formatter:off
+        /**
+         * Cache configuration file in XML whose path will be converted to URI to create the cache on first access.
+         * An example of the user defined property:
+         * quarkus.infinispan-client.cache.bookscache.configuration-uri=cacheConfig.xml
+         */
+        // @formatter:on
+        @ConfigItem
+        public Optional<String> configurationUri;
```

Comment:
I'm confused why we need both file and URI but I might be missing something. Could you explain why we would need both?

Reply:
We might not. This is why I moved the PR to draft. With the file I'm grabbing the content and I'm passing it into the configuration property. With the URI, I'm using the Infinispan URI remote config way. I was about to ask if there is a use case where using the URI or grabbing the content from a file inside the resources would cover different technical uses cases in different environments. If the answer is no, the file could be removed I guess